hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionUtils.java

```diff
@@ -196,13 +196,17 @@ static boolean isEmptyStopRow(byte[] row) {
     return Bytes.equals(row, EMPTY_END_ROW);
   }
 
-  static void resetController(HBaseRpcController controller, long timeoutNs, int priority) {
+  static void resetController(HBaseRpcController controller, long timeoutNs, int priority,
+    TableName tableName) {
     controller.reset();
     if (timeoutNs >= 0) {
       controller.setCallTimeout(
         (int) Math.min(Integer.MAX_VALUE, TimeUnit.NANOSECONDS.toMillis(timeoutNs)));
     }
     controller.setPriority(priority);
+    if (tableName != null) {
+      controller.setTableName(tableName);
```

Comment:
sorry i just caught this -- do you think we should setTableName with a null value when passed in? This would indicate a request that has no table, but if we don't set to null then it might attribute the request to an old table name since controllers are re-used. Alternatively, controller.reset() should reset the table name.

Reply:
@bbeaudreault hi, thanks for your suggestion, but actually I have a question. Is it possible for the table name to initially not be null, but later become null due to controller re-used? Or in other words, can the table name change for the same `RpcRetryingCaller`?  And I took a quick look at the code, and it seems that when the table name is null, it won't hit the `Get/Mutate/Scan/Multi`(ClientService) switch cases in the `MetricsConnection#updateRpc` method. Okay, I will reset the table name in the controller.reset() method, and keep everything else as it is. What do you think?

Reply:
@zhuyaogai I think you are right that its unlikely for the table name to change for the same RpcRetryingCaller. I'm just trying to ensure that we code defensively here, since who knows how the code will change in the future.

That said, the contract seems to be that we always should reset the controller before using it. So I think you're good to just reset the table name there, and leave this code here as-is.